files/build_templates/buffers_config.j2

```diff
@@ -128,6 +128,16 @@ def
 {%- endfor %}
 {%- set port_names_active  = port_names_list_active  | join(',') %}
 
+{%- set port_names_list_extra_queues = [] %}
+{%- for port in PORT_ACTIVE %}
+    {% if (generate_queue_buffers_with_extra_lossless_queues is defined or generate_queue_buffers_with_extra_lossless_queues_with_inactive_ports is defined) and 
+            ('type' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['type'] == 'LeafRouter' and DEVICE_NEIGHBOR_METADATA is defined and DEVICE_NEIGHBOR[port].name in DEVICE_NEIGHBOR_METADATA and DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[port].name].type == 'ToRRouter') or
+            ('subtype' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['subtype'] == 'DualToR' and DEVICE_NEIGHBOR_METADATA is defined and DEVICE_NEIGHBOR[port].name in DEVICE_NEIGHBOR_METADATA and DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[port].name].type == 'LeafRouter') %}
+        {%- if port_names_list_extra_queues.append(port) %}{%- endif %}
+    {% endif %}
+{%- endfor %}
+{%- set port_names_extra_queues = port_names_list_extra_queues | join(',') %}
+
 {%- set port_names_list_inactive  = [] %}
 {%- for port in PORT_INACTIVE %}
```

Comment:
you will need to have the same logic as 133 here as well when generate_queue_buffers_with_extra_lossless_queues_with_inactive_ports is defined, to find out the ports in the inactive list that need the extra queues

Reply:
I don't think it's a valid path.
Because the `PORT_INACTIVE` is the ports that are not in `DEVICE_NEIGHBOR`. So, we have no info to determine the type of neighbor. Hence, we don't know if extra queues are required because we need the neighbor's info.

Reply:
Makes sense. My bad